src/Microsoft.Identity.Web.OidcFIC/OidcIdpSignedAssertionProvider.cs

```diff
@@ -17,21 +19,58 @@ internal class OidcIdpSignedAssertionProvider : ClientAssertionProviderBase
         private readonly ITokenAcquirerFactory _tokenAcquirerFactory;
         private readonly MicrosoftIdentityApplicationOptions _options;
         private readonly string? _tokenExchangeUrl;
+        private readonly ILogger? _logger;
 
-        public OidcIdpSignedAssertionProvider(ITokenAcquirerFactory tokenAcquirerFactory, MicrosoftIdentityApplicationOptions options, string? tokenExchangeUrl)
+        public bool RequiresSignedAssertionFmiPath { get; internal set; }
+
+        public OidcIdpSignedAssertionProvider(ITokenAcquirerFactory tokenAcquirerFactory, MicrosoftIdentityApplicationOptions options, string? tokenExchangeUrl, ILogger? logger)
         {
             _tokenAcquirerFactory = tokenAcquirerFactory;
             _options = options;
             _tokenExchangeUrl = tokenExchangeUrl;
+            _logger = logger;
         }
 
         protected override async Task<ClientAssertion> GetClientAssertionAsync(AssertionRequestOptions? assertionRequestOptions)
         {
             _tokenAcquirer ??= _tokenAcquirerFactory.GetTokenAcquirer(_options);
 
             string tokenExchangeUrl = _tokenExchangeUrl ?? "api://AzureADTokenExchange";
+            AcquireTokenOptions? acquireTokenOptions = null;
+
+            // During the construction of the CCA, IdWeb tried to understand which Credential description to use and to skip, and thefore
+            // attempts to load the credentials with assertionRequestOptions = null (whereas it's not null when MSAL calls GetSignedAssertionAsync).
+            // If assertionRequestOptions = null and RequiresSignedAssertionFmiPath is true
+            // we postpone getting the signed assertion until the first call, when ClientAssertionFmiPath will be provided.
```

Comment:
Please log an issue to generalize this mechanism, as we discussed.

Reply:
https://github.com/AzureAD/microsoft-identity-web/issues/3394